packages/health/lib/src/health_factory.dart

```diff
@@ -16,9 +16,15 @@ class HealthFactory {
   String? _deviceId;
   final _deviceInfo = DeviceInfoPlugin();
   HealthLogger? _logger;
+  late bool _useHealthConnectIfAvailable;
```

Comment:
Actually it can be removed from here, as we just need to pass it in the constructor, but not store it in class variable. This flag should be passed to the initiation of the native health plugin.